clean_build.sh

```diff
@@ -1,32 +1,74 @@
 #!/bin/sh
+
 cd "$(dirname "$0")" || exit 1
+
 # some environments have cmake v2 as 'cmake' and v3 as 'cmake3'
 # check for cmake3 first then fallback to just cmake
-B_CMAKE=`type cmake3 2>/dev/null`
-if [ $? -eq 0 ]; then
-    B_CMAKE=`echo "$B_CMAKE" | cut -d' ' -f3`
+
+B_CMAKE=$(command -v cmake3 2>/dev/null)
+if [ "$?" -eq 0 ]; then
+    # Continue, cmake3 exists in $PATH.
+    continue
 else
-    B_CMAKE=cmake
+    # OK, so cmake3 isn't in path, so let's test to see if `cmake` itself exists, before proceeding.
+    if command -v cmake 2>/dev/null; then
+        B_CMAKE=$(command -v cmake)
+        # We have a cmake executable available, now let's proceed!
+    else
+        # As self-explanatory, the cmake executable isn't available, so we should fail here.
+        echo "ERROR: CMake not in $PATH, cannot build! Please install CMake, or if this persists, file a bug report."
+        exit 1
+    fi
 fi
-# default build configuration
+
+# Set default build type, but in CI, this is set to Release.
+# For downstream distributions, our recommendation is to set this type in your build scripts to Release, and supply Debug build types in your debug packages.
```

Comment:
I doubt downstreams will actually use this script, it's just asking for trouble and additional risk in case we break something here.

Reply:
For example, we do network access in the script, which is usually big no no and build machines have network completely disabled. If they did allow network access then reproducible builds would be much harder and there would be risk in malware infection because the code that runs on build machine is external.

Reply:
Also removed, good point about reproducible builds.

Reply:
Fixed in a1bc7a32.